src/pybind/mgr/devicehealth/module.py

```diff
@@ -526,7 +526,7 @@ def put_device_metrics(self, devid: str, data: Any) -> None:
                 self.set_device_wear_level(devid, -1.0)
 
     def _t2epoch(self, t: Optional[str]) -> int:
-        if t is None:
+        if not t:
```

Comment:
With this change I don't think the SQL changes are necesary, right?

Reply:
I think they are, or else you end up with ``(time = ? OR ? <= time)`` -> ``(time = 123 OR 0 <= time)`` which is everything instead of just 123.

Reply:
Hm, I don't know how this worked in my testing when I made this change. Anyway, that makes sense to me.